test/libsolidity/semanticTests/various/code_length_contract_member.sol

```diff
@@ -0,0 +1,17 @@
+// Test to see if type.code.length does extcodesize(type) only when type is an address.
+struct S {
+    bytes32 code;
+    bytes32 another;
+}
+
+contract C {
+    S s;
+
+    function f() public returns (uint, uint, bool) {
+        return (s.code.length, s.another.length, address(this).code.length > 150);
+    }
+}
+// ====
+// compileViaYul: also
```

Comment:
Hmm.. looks like this works already via Yul. I would have expected that something should have been changed in `libsolidity/codegen/ir/IRGeneratorForStatements.cpp` too. Why is it working? Looks like that I missed something.

Reply:
It does work, but `address.code.length`, does something like

```
size := extcodesize(address)
codecopy(....)
ans := sub(...)
```

I should also implement the shortcut, to do just and remove the redundant `codecopy`

```
ans := extcodesize(address)
``